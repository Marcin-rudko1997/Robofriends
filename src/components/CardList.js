import React from 'react';
import Card from './Card/Card';

const cardList = ({robots, onDeletedRobot}) => {
    //console.log('Card list');
    const cardsArray = robots.map((user, index ) =>  {
        return <Card key={index} id={user.id} name={user.name} email={user.email} deleteRobot={() => onDeletedRobot(user.id)}/>
    })
    return(
        <div>
            {cardsArray}
        </div>
        
    );
}

export default cardList;