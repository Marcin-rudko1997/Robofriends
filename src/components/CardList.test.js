import React from 'react';
import {shallow} from 'enzyme';
import CardList from './CardList';

it('renders CardList component', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'John John',
            email: 'John@gmail.com'
        }
    ]
    expect(shallow(<CardList robots={mockRobots}/>)).toMatchSnapshot();
})