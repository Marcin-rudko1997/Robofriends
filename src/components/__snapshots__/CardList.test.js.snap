// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <cardList
    robots={
        Array [
            Object {
              "email": "John@gmail.com",
              "id": 1,
              "name": "John John",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <card
          deleteRobot={[Function]}
          email="John@gmail.com"
          id={1}
          name="John John"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "deleteRobot": [Function],
          "email": "John@gmail.com",
          "id": 1,
          "name": "John John",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <card
            deleteRobot={[Function]}
            email="John@gmail.com"
            id={1}
            name="John John"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "deleteRobot": [Function],
            "email": "John@gmail.com",
            "id": 1,
            "name": "John John",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
